src/dqrobotics/interfaces/vrep/DQ_VrepInterface.cpp

```diff
@@ -566,3 +590,531 @@ bool DQ_VrepInterface::is_video_recording()
+ * @param function_name The name of the script function to call in the specified script.
+ * @param link name. The name of the object from which we want to extract the mass.
+ * @returns The mass of the object.
+ *
+ *              Example:
+ *              // This example assumes that in your CoppeliaSim there is a child script called
+ *              // "DQRoboticsApiCommandServer", where is defined the following Lua function:
+ *              //
+ *              //  function get_mass(inInts,inFloats,inStrings,inBuffer)
+ *              //     local mass = {}
+ *              //     if #inInts>=1 then
+ *              //         mass =sim.getShapeMass(inInts[1])
+ *              //         return {},{mass},{},''
+ *              //     end
+ *              //  end
+ *              // In addition, it is assumed that there is a FrankaEmikaPanda robot manipulator
+ *              // in the Coppelia scene.
+ *
+ *              DQ_VrepInterface vi;
+ *              double mass = vi.extract_mass("DQRoboticsApiCommandServer","get_mass","Franka_link2_resp");
+ *
+ */
+double DQ_VrepInterface::extract_mass(const std::string& obj_name, const std::string& function_name, const std::string&link_name)
+
+{
+    struct call_script_data data;
+    int my_handle = get_object_handle(link_name);
+    data = call_script_function(obj_name, function_name, {my_handle}, {}, {});
+    int size = data.output_floats.size();
+    if (size != 1){
+        throw std::range_error("Error in extract_center_of mass. Incorrect number of returned values from CoppeliaSim. (Expected: 1)");
+    }
+    return data.output_floats[0];
+}
+
+
+/**
+ * @brief This protected method calls remotely a CoppeliaSim script function.
+ * @param obj_name The name of the object where the script is attached to.
+ * @param scripttype The type of the script.
+ * @param function_name The name of the script function to call in the specified script.
+ * @param input_ints The input integer values.
+ * @param input_floats The input float values.
+ * @param input_strings The input string values.
+ * @param opmode The operation mode.
+ * @returns a call_script_data structure.
+ *
+ */
+call_script_data DQ_VrepInterface::_call_script_function(const std::string&  obj_name, const SCRIPT_TYPES& scripttype, const std::string&  function_name,
+                                                          const std::vector<int>& input_ints, const std::vector<float>& input_floats,
+                                                          const std::vector<std::string> &input_strings, const OP_MODES& opmode)
+{
+    struct call_script_data data;
+    int return_code = 1;
+    VectorXi  vec_output_ints = VectorXi::Zero(1);
+    VectorXd  vec_output_floats = VectorXd::Zero(1);
+    std::vector<std::string> vec_output_strings;
+
+
+    int outFloatCnt;
+    float* output_floats;
+
+    int outIntCnt;
+    int* output_ints;
+
+    int outStringCnt;
+    char* output_strings;
+
+    int* input_ints_ptr = nullptr;
```

Comment:
This can be reduced to this
```cpp
int* input_ints_ptr = input_ints.data()
```
Naturally, it can be used directly in the function you're wrapping without having to go through all these loops.
https://en.cppreference.com/w/cpp/container/vector/data

Reply:
Indeed!, your suggestion reduced the code a lot! 
Thank you!